core/src/test/scala/org/apache/spark/util/ThreadUtilsSuite.scala

```diff
@@ -25,11 +25,27 @@ import scala.concurrent.duration._
 import scala.util.Random
 
 import org.scalatest.concurrent.Eventually._
+import org.slf4j.MDC
 
 import org.apache.spark.SparkFunSuite
 
 class ThreadUtilsSuite extends SparkFunSuite {
 
+  test("newDaemonSingleThreadExecutor with MDC") {
+    val appIdValue = s"appId-${System.currentTimeMillis()}"
+    MDC.put("appId", appIdValue)
```

Comment:
I'm still worried about API inconsistency. We should use either `sc.setLocalProperties` or `MDC.put` for both driver and executor side MDC properties.

Reply:
@cloud-fan I think he is using `MDC.put` in both cases. The only difference is that for executor we also have to call `sc.setLocalProperties` in order to pass the property into the executor JVM. Once the executor process has knowledge of the appId or any other property, it is put into MDC the same way.

Reply:
I mean user-facing APIs. Eventually, we have to call `MDC.put` as this is how MDC works.

Maybe it's better to say that Spark can propagate the MDC properties to both driver thread pools and executor tasks. So end-users only need to call `MDC.put`.

Reply:
For the driver is true. for the Executor as it is in another process it should be called explicitly by the user.

Reply:
right. there is no way to make MDC put something in another process.